src/App.js

```diff
@@ -0,0 +1,113 @@
+import WaitingList from './WaitingList.js';
+import FinishedList from './FinishedList.js';
+import AddTodoBtn from './AddTodoBtn.js';
+
+function App({app, initialState, todoID}) {
+
+    this.todoID = localStorage.getItem('todoID') ? JSON.parse(localStorage.getItem('todoID')) : todoID;
+    this.state = {
+        waitingTodo: localStorage.getItem('waitingTodo') ? JSON.parse(localStorage.getItem('waitingTodo')) : initialState.waitingTodo,
+        finishedTodo: localStorage.getItem('finishedTodo') ? JSON.parse(localStorage.getItem('finishedTodo')) : initialState.finishedTodo,
+    };
+    this.setState = (nextState) => {
+
+        this.state = {
+            ...this.state,
+            ...nextState
+        }
+
+        localStorage.setItem('todoID', JSON.stringify(this.todoID));
+        localStorage.setItem('waitingTodo', JSON.stringify([...this.state.waitingTodo]));
+        localStorage.setItem('finishedTodo', JSON.stringify([...this.state.finishedTodo]));
+
+        this.render();
+    }
```

Comment:
localStorage를 모든 state value에 대해서 키와 value쌍을 만드는 것보다
전역 state 객체를 하나로 관리하고 setState가 일어날때마다 localStorage와 동기화시키는 건 어떨까요?

```jsx
function syncWithStorage(state) {
  LocalStorageAdapter.write('state', state);
}
```

```suggestion
    this.setState = (nextState) => {

        this.state = {
            ...this.state,
            ...nextState
        }
        
        syncWithStorage(this.state);
        this.render();
    }
```

Reply:
오,, 이렇게 하면 확실히 깔끔해지겠네요! 감사합니당